### PR TITLE
Fixed `AutoloadSourceLocatorTest::testCanAutoloadPsr4ClassesInPotentiallyMultipleDirectories()` autoloading of PHPUnit assertion classes

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php [hunk 1]
```diff
@@ -385,7 +385,25 @@ public function testReturnsNullWithInternalFunctions(): void
 
     public function testCanAutoloadPsr4ClassesInPotentiallyMultipleDirectories(): void
     {
-        spl_autoload_register([$this, 'autoload']);
+        // Force autoload for these asserts because it's not possible to autoload them in the autoload callback
```
Comment (Ocramius) [inline test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php:388]: This stuff should be done in `bootstrap.php`, no?

Comment (kukulich) [reply test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php:388]: Yes, the composer autoloader is in `bootstrap.php` but it does not work here ¯\_(ツ)_/¯  I tried `$prepend = true` for this autoload but it does not work either.

Comment (kukulich) [reply test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php:388]: To reproduce: `vendor/bin/phpunit test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php --filter=testCanAutoloadPsr4ClassesInPotentiallyMultipleDirectories`

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Pull request merged**